src/test/sv2_connman_tests.cpp

```diff
@@ -0,0 +1,201 @@
+#include <boost/test/unit_test.hpp>
+#include <sv2/connman.h>
+#include <sv2/messages.h>
+#include <sv2/transport.h>
+#include <test/util/net.h>
+#include <test/util/setup_common.h>
+#include <util/sock.h>
+
+#include <memory>
+
+BOOST_FIXTURE_TEST_SUITE(sv2_connman_tests, TestChain100Setup)
+
+/**
+  * A class for testing the Sv2Connman. Each ConnTester encapsulates a
+  * Sv2Connman (the one being tested) as well as a Sv2Cipher
+  * to act as the other side.
+  */
+class ConnTester : Sv2EventsInterface {
+private:
+    std::unique_ptr<Sv2Transport> m_remote_transport; //!< Transport for peer
+    // Sockets that will be returned by the Sv2Connman's listening socket Accept() method.
+    std::shared_ptr<DynSock::Queue> m_sv2connman_accepted_sockets{std::make_shared<DynSock::Queue>()};
+
+    std::shared_ptr<DynSock::Pipes> m_current_client_pipes;
+
+    XOnlyPubKey m_connman_authority_pubkey;
+
+public:
+    std::unique_ptr<Sv2Connman> m_connman; //!< Sv2Connman being tested
+
+    ConnTester()
+    {
+        CreateSock = [this](int, int, int) -> std::unique_ptr<Sock> {
+            // This will be the bind/listen socket from m_connman. It will
+            // create other sockets via its Accept() method.
+            return std::make_unique<DynSock>(std::make_shared<DynSock::Pipes>(), m_sv2connman_accepted_sockets);
+        };
+
+        CKey static_key;
+        static_key.MakeNewKey(true);
+        auto authority_key{GenerateRandomKey()};
+        m_connman_authority_pubkey = XOnlyPubKey(authority_key.GetPubKey());
+
+        // Generate and sign certificate
+        auto now{GetTime<std::chrono::seconds>()};
+        uint16_t version = 0;
+        // Start validity a little bit in the past to account for clock difference
+        uint32_t valid_from = static_cast<uint32_t>(std::chrono::duration_cast<std::chrono::seconds>(now).count()) - 3600;
+        uint32_t valid_to =  std::numeric_limits<unsigned int>::max(); // 2106
+        Sv2SignatureNoiseMessage certificate{version, valid_from, valid_to, XOnlyPubKey(static_key.GetPubKey()), authority_key};
+
+        m_connman = std::make_unique<Sv2Connman>(TP_SUBPROTOCOL, static_key, m_connman_authority_pubkey, certificate);
+
+        BOOST_REQUIRE(m_connman->Start(this, "127.0.0.1", 18447));
```

Comment:
How are you able to Start() the connman but never break out of `ThreadSocketHandler()` with `interruptNet()`? If I leave the thread running in my HTTP unit test I get an error when the SockMan instance is destroyed.

Reply:
No idea :-)

cc @vasild?

Reply:
The destructor of `Sv2Connman` calls the `Interrupt()` method which signals the `interruptNet` variable.